lib/src/native/realm_core.dart

```diff
@@ -1811,59 +1888,64 @@ class ConfigHandle extends HandleBase<realm_config> {
 }
 
 class RealmHandle extends HandleBase<shared_realm> {
+  int _counter = 0;
+
+  final Map<int, WeakReference<RootedHandleBase>> _children = {};
+
   RealmHandle._(Pointer<shared_realm> pointer) : super(pointer, 24);
 
   RealmHandle._unowned(Pointer<shared_realm> pointer) : super.unowned(pointer);
+
+  int addChild(RootedHandleBase child) {
+    final id = _counter++;
```

Comment:
Why not to keep these handles only in case the root is unowned `if (this.isUnowned)`. I think we musn't keep the RealmObjects pointers for Realms that are not unowned.

Reply:
We already do that - e.g.:

https://github.com/realm/realm-dart/blob/19107dca53780a2806e148852fd963b6574fe58d/lib/src/native/realm_core.dart#L1924

The object/result/list handle will set their `root` to `null` if the Realm is not unowned.